### PR TITLE
Box: Create elevation section

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Make our elevation color options more clear, link to elevation page. [Preview](https://deploy-preview-2037--gestalt.netlify.app/box#Elevation)
Also adds a border style for errors that was missed in a previous PR

#### Why?

Help clarify when to use our elevation colors. Also creates a space to add our future raised elevation shadow tokens with an explanation

### CODE REVIEW

File: docs/pages/box.js [hunk 1]
```diff
@@ -459,6 +456,16 @@ function Example() {
             {(props) => <Box width={60} height={60} rounding="circle" color={props.color} />}
           </CombinationNew>
         </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Elevation"
```
Comment (ayeshakmaz) [inline docs/pages/box.js:460]: new section around Elevation (will also house the new shadow options once those have been added)

File: packages/gestalt-design-tokens/tokens/color/alias.json [hunk 2]
```diff
@@ -243,12 +243,17 @@
...
-        "comment": "Used to delineate a larger container, like a card"
+        "comment": "Used to delineate a larger container, like a card or TextArea"
       },
       "default": {
         "value": "{color.gray.roboflow.500.value}",
         "darkValue": "{color.gray.roboflow.300.value}",
-        "comment": "Used to indicate specific UI element boundaries, like a TextField or SelectList"
+        "comment": "Used to indicate specific UI element boundaries, like a Checkbox or RadioButton"
+      },
+      "error": {
```
Comment (ayeshakmaz) [inline packages/gestalt-design-tokens/tokens/color/alias.json:253]: Adding this option, missed it with our last border updated

File: docs/pages/box.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/pages/box.js:460]: Could we add a background to simulate dark mode treatment? 


image
fc-bf71-71792473e001.png)

File: docs/pages/box.js [hunk 3]
```diff
@@ -459,6 +456,16 @@ function Example() {
...
+          description="These colors can elevate elements within the UI. In light mode, `elevationAccent` can be used when shadows or borders are not an option. `elevationFloating` and `elevationRaised` are only applicable in dark mode. For full details, visit our [Elevation foundations page](/elevation)."
+        >
+          <CombinationNew color={['elevationAccent', 'elevationFloating', 'elevationRaised']}>
+            {/* eslint-disable-next-line react/prop-types */}
+            {(props) => <Box width={60} height={60} rounding="circle" color={props.color} />}
```
Comment (AlbertCarreras) [inline docs/pages/box.js:465]: nit

{({ color }) => <Box width={60} height={60} rounding="circle" color={color} />}

File: packages/gestalt-design-tokens/tokens/color/alias.json [hunk 4]
```diff
@@ -243,12 +243,17 @@
       "container": {
         "value": "{color.gray.roboflow.300.value}",
         "darkValue": "{color.gray.roboflow.600.value}",
-        "comment": "Used to delineate a larger container, like a card"
+        "comment": "Used to delineate a larger container, like a card or TextArea"
```
Comment (AlbertCarreras) [inline packages/gestalt-design-tokens/tokens/color/alias.json:246]: nit question

like a **C**?ard or TextArea

File: packages/gestalt-design-tokens/tokens/color/alias.json [hunk 5]
```diff
@@ -243,12 +243,17 @@
...
       },
       "default": {
         "value": "{color.gray.roboflow.500.value}",
         "darkValue": "{color.gray.roboflow.300.value}",
-        "comment": "Used to indicate specific UI element boundaries, like a TextField or SelectList"
+        "comment": "Used to indicate specific UI element boundaries, like a Checkbox or RadioButton"
```
Comment (AlbertCarreras) [inline packages/gestalt-design-tokens/tokens/color/alias.json:251]: nit:

like Checkbox or RadioButton?

File: docs/pages/box.js (hunk 1 above)
Comment (ayeshakmaz) [reply docs/pages/box.js:460]: I'm actually going to update this in the other PR when I add the border options to this section

**Pull request merged**